src/item.cpp

```diff
@@ -3295,39 +3295,48 @@ void item::final_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
     }
 
     // list recipes you could use it in
-    itype_id tid;
-    if( contents.empty() ) { // use this item
-        tid = typeId();
-    } else { // use the contained item
-        tid = contents.front().typeId();
-    }
-    const std::set<const recipe *> &known_recipes = g->u.get_learned_recipes().of_component( tid );
-    if( !known_recipes.empty() && parts->test( iteminfo_parts::DESCRIPTION_APPLICABLE_RECIPES ) ) {
-        const inventory &inv = g->u.crafting_inventory();
+    if( parts->test( iteminfo_parts::DESCRIPTION_APPLICABLE_RECIPES ) ) {
+        itype_id tid;
+        if( contents.empty() ) { // use this item
+            tid = typeId();
+        } else { // use the contained item
+            tid = contents.front().typeId();
+        }
 
-        if( known_recipes.size() > 24 ) {
-            insert_separation_line( info );
-            info.push_back( iteminfo( "DESCRIPTION",
-                                      _( "You know dozens of things you could craft with it." ) ) );
-        } else if( known_recipes.size() > 12 ) {
+        // FIXME: Somehow the following lines are causing a game crash whenever the crafting menu is opened
+        const inventory &crafting_inv = g->u.crafting_inventory();
+        const std::set<const recipe *> &item_recipes = g->u.get_available_recipes( crafting_inv,
+                nullptr ).of_component( tid );
```

Comment:
This is a common pitfall. `of_component()` returns a reference to the result of `get_available_recipes()` which happens to be a temporary because nothing references it. It gets destroyed soon after being constructed. Just assign it to a variable, and it will be fine.
```c++
const recipe_subset available_recipe_subset = g->u.get_available_recipes( crafting_inv );
const std::set<const recipe *> &item_recipes = available_recipe_subset.of_component( tid );
```

Reply:
Many thanks! That got it working. Final commits on the way...

Reply:
No worries. I remember the first time I fell into that trap: spent hours debugging.